src/utils/files.ts

```diff
@@ -4,12 +4,14 @@ import { exit } from "process";
 import { BLOCK_SETTINGS_FILE, RC_FILE_PATH } from "../constants";
 import { formatName, logError, toPascalCase } from "./index";
 
-interface BlockFileObject {
+export interface BlockFileObject {
     category?: string;
     displayName: string;
-    id?: string;
+    id: string;
```

Comment:
Isn't this optional when we haven't publish the block the first time?

Reply:
yea, good call

Reply:
Okay, so I remember why I removed the optional. It was causing type issues for all the request interfaces that needed ID to exist. The block file interface doesn't get used until we know for a fact the ID has been inserted so it doesn't have any issues when removing the optional case.